pkg/profiler/cpu/maps.go

```diff
@@ -297,11 +300,18 @@ const (
 	pyperfModule
 )
 
-func initializeMaps(logger log.Logger, reg prometheus.Registerer, byteOrder binary.ByteOrder, modules map[moduleType]*bpf.Module) (*bpfMaps, error) {
+func initializeMaps(logger log.Logger, reg prometheus.Registerer, byteOrder binary.ByteOrder, arch string, modules map[moduleType]*bpf.Module) (*bpfMaps, error) {
 	if modules[nativeModule] == nil {
 		return nil, fmt.Errorf("nil nativeModule")
 	}
 
+	var compactUnwindRowSizeBytes int
+	if arch == "arm64" {
+		compactUnwindRowSizeBytes = compactUnwindRowSizeBytesArm64
+	} else {
+		compactUnwindRowSizeBytes = compactUnwindRowSizeBytesX86
+	}
```

Comment:
Potentially we can add some error in case the arch is not what we expect (if someone builds the agent in some other architecture)